src/React/Basic.purs

```diff
@@ -35,3 +38,50 @@ react
      }
   -> ReactComponent props
 react { initialState, render } = react_ { initialState, render: mkFn3 render }
+
+
+foreign import reactStatefulComponent_
```

Comment:
Could this wrap `react_` instead of adding another FFI function? 

I think we'd want `receiveProps` to be supported by neither or both anyway.

Reply:
Yep, sounds good.

On a related note, a `receiveProps`-like function is actually [going to become a React lifecycle function](https://twitter.com/dan_abramov/status/953448757953261568) at some point.